Sources/Plugins/SRTMPlugin/TerrainMode.swift

```diff
@@ -0,0 +1,239 @@
+//
+//  TerrainMode.swift
+//  OsmAnd Maps
+//
+//  Created by Skalii on 19.06.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc
+@objcMembers
+final class TerrainMode: NSObject {
+
+    @objc enum TerrainType: Int32 {
+        case hillshade
+        case slope
+        case height
+
+        var name: String {
+            switch self {
+            case .hillshade: "hillshade"
+            case .slope: "slope"
+            case .height: "height"
+            }
+        }
+    }
+
+    @objcMembers
+    final class TerrainTypeWrapper: NSObject {
+
+        static func getNameFor(type: TerrainType) -> String {
+            type.name
+        }
+    }
+
+    static let defaultKey = "default"
+    static let altitudeDefaultKey = "altitude_default"
+    static let hillshadePrefix: String = "hillshade_main_"
+    static let hillshadeScndPrefix = "hillshade_color_"
+    static let colorSlopePrefix = "slope_"
+    static let heightPrefix = "height_"
+
+    let type: TerrainType
+
+    private let minZoom: OACommonInteger
+    private let maxZoom: OACommonInteger
+    private let transparency: OACommonInteger
+    private let key: String
+
+    static var values: [TerrainMode] {
+        if let terrainModes {
+            return terrainModes
+        }
+        Self.reloadTerrainModes()
+        return terrainModes ?? []
+    }
+
+    private static var terrainModes: [TerrainMode]?
+
+    var translateName: String
+
+    init(_ key: String, type: TerrainType, translateName: String) {
+        self.key = key
+        self.type = type
+        self.translateName = translateName
+
+        let settings = OAAppSettings.sharedManager()!
+        minZoom = settings.registerIntPreference(key + "_min_zoom", defValue: 3).makeProfile()
+        maxZoom = settings.registerIntPreference(key + "_max_zoom", defValue: 17).makeProfile()
+        transparency = settings.registerIntPreference(key + "_transparency", defValue: type == .hillshade ? 100 : 80).makeProfile()
+    }
+
+    static func reloadTerrainModes() {
+        var modes = [TerrainMode]()
+        modes.append(TerrainMode(defaultKey, type: .hillshade, translateName: localizedString("shared_string_hillshade")))
+        modes.append(TerrainMode(defaultKey, type: .slope, translateName: localizedString("shared_string_slope")))
+        
+        if let dir = OsmAndApp.swiftInstance().colorsPalettePath,
+           let files = try? FileManager.default.contentsOfDirectory(atPath: dir) {
+            for file in files {
+                guard file.hasSuffix(TXT_EXT) else { continue }
+
+                let nm = file
+                if nm.hasPrefix(hillshadePrefix) {
```

Comment:
It seems we can simplify by extracting the implementation into a separate method and iterating through a dictionary like [.hillshade: hillshadePrefix, ...]